Sparkle/SUSignatures.m

```diff
@@ -45,10 +55,8 @@ - (const unsigned char *)ed25519Signature {
 // Xcode may enable this in pedantic mode
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Wdirect-ivar-access"
-    for(size_t i=0; i < sizeof(self->ed25519_signature); i++) {
-        if (self->ed25519_signature[i] != 0) {
-            return self->ed25519_signature;
-        }
+    if (self->has_ed25519_signature) {
+        return self->ed25519_signature;
```

Comment:
that could return all-0xFF signature. It doesn't seem right.

Reply:
An all-0xFF signature is always invalid per `ed25519_verify`, so it's safe to return as something that will always fail.